syntactic-paths/src/main/java/com/palantir/util/syntacticpath/Path.java

```diff
@@ -102,15 +99,25 @@ private static String checkCharacters(String path) {
     }
 
     private static List<String> checkSegments(List<String> segments) {
-        if (!Collections.disjoint(segments, ILLEGAL_SEGMENTS)) {
+        if (containsIllegalSegment(segments)) {
             throw new SafeIllegalArgumentException(
                     "Path contains illegal segments", UnsafeArg.of("segments", segments));
+        } else {
+            return segments;
         }
-        return segments;
+    }
+
+    private static boolean containsIllegalSegment(Iterable<String> iterable) {
+        for (String segment : iterable) {
+            if (Path.ILLEGAL_SEGMENTS.contains(segment)) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
I think this can be optimized to the following, given there's only one illegal segment:
```suggestion
    private static boolean containsIllegalSegment(List<String> input) {
        return input.contains(".");
```
trade-off is that if we added another illegal segment in the future, we'd have to bring back the loop (or more likely, add an 'or' condition)

Reply:
good call, updated.